### PR TITLE
add sourmash rules

### PR DESCRIPTION
**Pull request by RNAer**

### CODE REVIEW

**Review by tanaes — APPROVED** [review]
This looks great, thanks!

File: oecophylla/distance/sourmash.rule [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    benchmark:
+        "benchmarks/distance/sourmash_dm.txt"
+    run:
+        from skbio.stats.distance import DistanceMatrix
+        from skbio.io import write
+        import pandas as pd
+        import numpy as np
+        shell('''
+            set +u; {params.env}; set -u
+            sourmash compare {i} --csv {output[0]}'.format(i=' '.join(input))
```
Comment (tkosciolek) [inline oecophylla/distance/sourmash.rule]: in `shell()` you don't need to specify `{output[0]}`, it should be simply `{output}`

Comment (tkosciolek) [reply oecophylla/distance/sourmash.rule]: also how does that work with this Python syntax? `.format()` ?

File: oecophylla/distance/sourmash.rule [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    log:
+        distance_dir + "logs/sourmash_sig.sample=[{sample}].log"
+    benchmark:
+        "benchmarks/distance/sourmash_sig.sample=[{sample}].txt"
+    run:
+        with tempfile.TemporaryDirectory(dir=TMP_DIR_ROOT) as temp_dir:
+            shell("""
+                  set +u; {params.env}; set -u
+                  zcat {input.forward} {input.reverse} > {temp_dir}/{wildcards.sample}
+                  sourmash compute --scaled {params.scaled} -k {params.kmer} -o {output[0]}
```
Comment (tkosciolek) [inline oecophylla/distance/sourmash.rule:25]: `-o` should be `{output}`, not `{output[0]}`

File: oecophylla/distance/sourmash.rule [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        1
+    log:
+        distance_dir + "logs/sourmash_sig.sample=[{sample}].log"
+    benchmark:
+        "benchmarks/distance/sourmash_sig.sample=[{sample}].txt"
+    run:
+        with tempfile.TemporaryDirectory(dir=TMP_DIR_ROOT) as temp_dir:
+            shell("""
+                  set +u; {params.env}; set -u
+                  zcat {input.forward} {input.reverse} > {temp_dir}/{wildcards.sample}
```
Comment (tkosciolek) [inline oecophylla/distance/sourmash.rule:24]: where does `{wildcards.sample}` come from?

**Comment by RNAer** [comment]
I have 2 output files specified. need index to provide 1st one

**Review by tkosciolek — COMMENTED** [review]
some minor-ish comments

File: oecophylla/distance/sourmash.rule (hunk 3 above)
Comment (RNAer) [reply oecophylla/distance/sourmash.rule:24]: that the syntax of snakemake to get the value of `{sample}` given in input

File: oecophylla/distance/sourmash.rule (hunk 1 above)
Comment (RNAer) [reply oecophylla/distance/sourmash.rule]: right. just fixed

File: oecophylla/distance/sourmash.rule (hunk 2 above)
Comment (RNAer) [reply oecophylla/distance/sourmash.rule:25]: better to be explicit?

**Comment by RNAer** [comment]
thanks, @tkosciol . I think I addressed all your comments? @tkosciol @tanaes , feel free to merge.

**Pull request merged**